### DISCUSSION TITLE
Calendar: Add support of new parameter "displayDateFormat", and keep "dateFormat" as format of technical date value

### DISCUSSION THREAD

**Discussion by dreyliky** [primefaces/community#1095] [PrimeNG] [answered]
I think it is a common task and I don't understand why I can't define one format for technical date value, and another format for the date which displayed in the input element.

I see this API like this:
```
<p-calendar
      [displayDateFormat]="'mm/dd/yy'"
      [dateFormat]="'yy-mm-dd'"
      [showIcon]="true"
      [showButtonBar]="true"
      [firstDayOfWeek]="1"
      [dataType]="'string'"
      [formControlName]="'date'"
></p-calendar>
```

I already implemented this in my local directive by monkey patching original component logic like this, so it is not complicated to implement but so useful for many people I believe:
https://stackblitz.com/edit/n1yaqc?file=src%2Fapp%2Fdemo%2Fcalendar-reactive-forms-demo.html

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We'll wait for community upvotes to implement this feature.

**Comment by MartinKoeditz** [comment]
I need this feature too. I have the same problem.

**Comment by tambor81** [comment]
Same this is useful !!!!

We need to add this, plus the ability to add more decorators at the beginning and at the end.

in selectionMode="range"     we only have DateSeparator, it is useful but I would need to decorate it to see something like:

From:   {{date.minDate}  |   To:    {{date.maxDate}}  selected.


currently we can only use DateSeparator for changing the "-" to be   "| To: ",  
however it is impossible to add "From:"  at the beginning or  " selected" at the end

**Comment by e-marfoeldi-carnect** [comment]
The example code above gave me an idea for my case:

PrimeNG (v17)

```typescript
@Directive({
  selector: 'p-calendar[calendarValue]',
  standalone: true,
})
export class CalendarValueDirective {
  constructor(@Self() private readonly calendar: Calendar) {
    const originalUpdateModel = this.calendar.updateModel.bind(this.calendar);
    const originalParseDateTime = this.calendar.parseDateTime.bind(
      this.calendar
    );

    this.calendar.parseDateTime = (text: string) => {
      const date =
        this.calendar.dataType === 'iso8601'
          ? DateConverter.deserialize(text)
          : null;

      return date ?? originalParseDateTime(text);
    };

    this.calendar.updateModel = (value: unknown) => {
      if (this.calendar.dataType === 'iso8601' && value instanceof Date) {
        this.calendar.value = value;
        if (this.calendar.isSingleSelection()) {
          this.calendar.onModelChange(DateConverter.serialize(value) ?? null);
        } else {
          const isoArrValue = Array.isArray(value)
            ? value.map((date: Date) => DateConverter.serialize(date) ?? null)
            : null;
          this.calendar.onModelChange(isoArrValue);
        }
      } else {
        originalUpdateModel(value);
      }
    };
  }
}
```
